private-templates-service/adaptivecards-templating-service/src/storageproviders/MongoDBProvider.ts

```diff
@@ -0,0 +1,208 @@
+    return await this.Template.create(template)
+      .then(result => {
+        return Promise.resolve({ success: true, result: 1 });
+      })
+      .catch(e => {
+        return Promise.resolve({
+          success: false,
+          number: 0,
+          errorMessage: e
+        });
+      });
+  }
+  async removeUser(query: Partial<IUser>): Promise<JSONResponse<Number>> {
+    return await this.User.deleteOne(query)
+      .then(result => {
+        if (result.deletedCount) {
+          return Promise.resolve({
+            success: true,
+            result: result.deletedCount
+          });
+        }
+        return Promise.resolve({
+          success: false,
+          result: 0,
+          errorMessage: "No users found matching given criteria"
+        });
+      })
+      .catch(e => {
+        return Promise.resolve({
+          success: false,
+          number: 0,
+          errorMessage: e
+        });
+      });
+  }
+  async removeTemplate(query: Partial<ITemplate>): Promise<JSONResponse<Number>> {
+    return await this.Template.deleteOne(query)
+      .then(result => {
+        if (result.deletedCount) {
+          return Promise.resolve({
+            success: true,
+            result: result.deletedCount
+          });
+        }
+        return Promise.resolve({
+          success: false,
+          result: 0,
+          errorMessage: "No templates found matching given criteria"
+        });
+      })
+      .catch(e => {
+        return Promise.resolve({
+          success: false,
+          number: 0,
+          errorMessage: e
+        });
+      });
+  }
+
+  // Set up mongoose models to work with the chosen collections
+  async setUpCollections(): Promise<JSONResponse<Boolean>> {
+    try {
+      this.User = Mongo.Model.getCollection<IUserModel>(this.db, USERS_COLLECTION_NAME_SINGULAR, UserSchema);
+      this.Template = Mongo.Model.getCollection<ITemplateModel>(this.db, TEMPLATES_COLLECTION_NAME_SINGULAR, TemplateSchema);
+      return Promise.resolve({ success: true });
+    } catch (e) {
+      return Promise.resolve({ success: false, errorMessage: e });
+    }
+  }
+
+  async initializeDB(): Promise<JSONResponse<Boolean>> {
+    return await mongoose
+      .createConnection(this.connectionString, this.options)
+      .then(connection => {
+        this.db = connection;
+        return this.setUpCollections();
+      })
+      .catch(e => {
+        return Promise.resolve({ success: false, errorMessage: e });
+      });
+  }
+
+  async connect(): Promise<JSONResponse<Boolean>> {
+    return await this.initializeDB();
+  }
+
+  async close(): Promise<JSONResponse<Boolean>> {
+    return await this.db
+      .close()
+      .then(result => {
+        return Promise.resolve({ success: true });
+      })
+      .catch(e => {
+        return Promise.resolve({ success: false, errorMessage: e });
+      });
+  }
+
+  constructor(connectionString: string, options: any) {
```

Comment:
When you've moved this to it's separate file, put the constructor at the top.

Reply:
done :)